### ISSUE TITLE
Popups

### ISSUE THREAD

**Issue by brambow** [boundlessgeo/sdk#821]
I already mentioned the dependency on font-awesome in [#788](https://github.com/boundlessgeo/sdk/issues/788)

I recently ran into an issue where I tried to do 2 things, neither of which worked.

1) Wrap my custom popup (extended from SdkPopup) with Redux's connect in order to bring in an action via mapDispatchToProps.  This resulted in an error:

![image](https://user-images.githubusercontent.com/6227786/36275773-c4f7b39e-1259-11e8-9ed9-ee4880f971c5.png)

2) Wrap the same popup with Apollo-GraphQL's withRouter function (no longer trying to use redux connect). This resulted in an error:

![image](https://user-images.githubusercontent.com/6227786/36275835-fdc704f4-1259-11e8-8c4e-caaaf632db15.png)

In both cases, such errors are only happening with the Popup component.  All other components in the application work fine with the Redux and Apollo providers.

The Popups example makes use of an undocumented function map.addPopup(), so I checked it out.  Some things I noticed:

[Line 1087](https://github.com/boundlessgeo/sdk/blob/master/src/components/map.js#L1087) of the map component js file has this line:

`const elem = document.createElement('div');`

Which is direct DOM manipulation, a no-no in React, and I think gets to the crux of my issues.  The content of a popup is being rendering into a DOM that is separate from the rest of the application, and thus why my popup can't recognize the same Apollo provider that the rest of the application does.

After looking at this some, I'm considering trying to implement an alternative to the SdkPopup that uses a React portal instead of creating an openlayers Overlay.  But I'm not really sure how to attach it to the feature coordinates.  

Any feedback/suggestions on anything popup related would be appreciated!

**Comment by bartvde** [comment]
yes we are aware of the issue and it was reported before by @milafrerichs who said he could not use material-ui components inside of the popup because the MuiThemeProvider could not be used.

I've also contemplated using portal.

So as I unfortunately cannot provide much guidance, we would be very interested to see what you can come up with.
